Source/SwiftLintBuiltInRules/Rules/Style/OpeningBraceRule.swift

```diff
@@ -25,31 +25,123 @@ struct OpeningBraceRule: SwiftSyntaxCorrectableRule {
+            }
+
+            super.visitPost(node)
+        }
+
+        override func visitPost(_ node: ProtocolDeclSyntax) {
+            if configuration.ignoreMultilineTypeHeaders,
+                hasMultilinePredecessors(node.memberBlock, keyword: node.protocolKeyword) {
                 return
             }
-            if configuration.allowMultilineFunc, refersToMultilineFunction(body, functionIndicator: node.initKeyword) {
+
+            super.visitPost(node)
+        }
+
+        override func visitPost(_ node: StructDeclSyntax) {
+            if configuration.ignoreMultilineTypeHeaders,
+                hasMultilinePredecessors(node.memberBlock, keyword: node.structKeyword) {
                 return
             }
-            collectViolations(for: body)
+
+            super.visitPost(node)
         }
 
-        private func refersToMultilineFunction(_ body: CodeBlockSyntax, functionIndicator: TokenSyntax) -> Bool {
+        // MARK: - Conditional Statements
+
+        override func visitPost(_ node: ForStmtSyntax) {
+            if configuration.ignoreMultilineStatementConditions,
+                hasMultilinePredecessors(node.body, keyword: node.forKeyword) {
+                return
+            }
+
+            super.visitPost(node)
+        }
+
+        override func visitPost(_ node: IfExprSyntax) {
+            if configuration.ignoreMultilineStatementConditions,
+                hasMultilinePredecessors(node.body, keyword: node.ifKeyword) {
+                return
+            }
+
+            super.visitPost(node)
+        }
+
+        override func visitPost(_ node: WhileStmtSyntax) {
+            if configuration.ignoreMultilineStatementConditions,
+                hasMultilinePredecessors(node.body, keyword: node.whileKeyword) {
+                return
+            }
+
+            super.visitPost(node)
+        }
+
+        // MARK: - Functions and Initializers
+
+        override func visitPost(_ node: FunctionDeclSyntax) {
+            if let body = node.body,
+                configuration.shouldIgnoreMultilineFunctionSignatures,
+                hasMultilinePredecessors(body, keyword: node.funcKeyword) {
+                return
+            }
+
+            super.visitPost(node)
+        }
+
+        override func visitPost(_ node: InitializerDeclSyntax) {
+            if let body = node.body,
+                configuration.shouldIgnoreMultilineFunctionSignatures,
+                hasMultilinePredecessors(body, keyword: node.initKeyword) {
+                return
+            }
+
+            super.visitPost(node)
+        }
+
+        // MARK: - Other Methods
+
+        /// Checks if a `BracedSyntax` has a multiline predecessor.
+        /// For type declarations, the predecessor is the header. For conditional statements,
+        /// it is the condition list, and for functions, it is the signature.
+        private func hasMultilinePredecessors(_ body: some BracedSyntax, keyword: TokenSyntax) -> Bool {
```

Comment:
The name `Predecessors` is not too obvious. But I cannot think of anything better and the method is documented very well. So that's fine. 👍

Reply:
I agree, same feeling here... I'm not happy with the name but couldn't describe it better.